AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Actions/RefreshAuthorizationSession/InitializeRefreshSession.swift

```diff
@@ -0,0 +1,72 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+
+struct InitializeRefreshSession: Action {
+
+    let identifier = "InitializeRefreshSession"
+
+    let existingCredentials: AmplifyCredentials
+
+    let isForceRefresh: Bool
+
+    func execute(withDispatcher dispatcher: EventDispatcher, environment: Environment) {
+
+        logVerbose("\(#fileID) Starting execution", environment: environment)
+
+        let event: RefreshSessionEvent
+
+        switch existingCredentials {
+        case .userPoolOnly(let tokens):
+            event = .init(eventType: .refreshCognitoUserPool(tokens, nil))
```

Comment:
May be have a separate event for the use case where there is no identity ID..

Reply:
I added that to make the statemachine simple, but I think it should be separate, so created a new state and event. Updated the description as well.